### PR TITLE
Use plone i18n domain

### PR DESCRIPTION
**Pull request by staeff**
See https://github.com/plone/Products.CMFPlone/issues/983

### CODE REVIEW

File: plone/formwidget/autocomplete/__init__.py [hunk 1]
```diff
@@ -1,5 +1,4 @@
 from zope.i18nmessageid import MessageFactory
-MessageFactory = MessageFactory('plone.formwidget.autocomplete')
 
-from plone.formwidget.autocomplete.widget import AutocompleteFieldWidget
-from plone.formwidget.autocomplete.widget import AutocompleteMultiFieldWidget
```
Comment (vincentfretin) [inline plone/formwidget/autocomplete/__init__.py]: Why did you remove this? If it was there, it means it's a part of the public API. I'm sure some package import from here instead of widget. Please revert.

Comment (staeff) [inline plone/formwidget/autocomplete/__init__.py]: I didn't know that and thought it was some cruft left behind. I re-added it.

**Comment by mauritsvanrees** [comment]
@vincentfretin Plone 4.3 still uses the master branch of this package. But 4.3 uses the 4.3.x branch of plone.app.locales, which will not have the translations for this package. So I think this means we have to create a 4.3 compatible branch of plone.formwidget.autocomplete from before this merge and increase the version of master. Do you agree?

I can do this easily. Just checking if I am misjudging anything.

**Comment by vincentfretin** [comment]
Maybe we should include the messages in plone.app.locales 4.3.x as well, to avoid maintaining branch for plone.formwidget.autocomplete just for translations, and maybe other packages @staeff worked on recently.

**Comment by vincentfretin** [comment]
Follow up on https://github.com/plone/Products.CMFPlone/issues/983#issuecomment-181845631

**Comment by mauritsvanrees** [comment]
I have released version 1.2.9 with this, for 4.3 and 5.0. Thanks!
